File: 001-tdd-wprowadzenie/Java/src/test/java/lukaszstaniszewski/StrongCalcTest.java
package lukaszstaniszewski;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * The type Strong calc test.
 */
public class StrongCalcTest {

    private StrongCalc strongCalc;

    /**
     * Sets up.
     */
    @Before
    public void setUp() {
        this.strongCalc = new StrongCalc();
    }

    /**
     * Calc.
     *
     * @throws Exception the exception
     */
    @Test
    public void calc() throws Exception {
        assertEquals(6, this.strongCalc.calc(3));
    }

    /**
     * Negative strong.
     *
     * @throws Exception the exception
     */
    @Test(expected = Exception.class)
    public void negativeStrong() throws Exception {
        this.strongCalc.calc(-3);
    }
}

File: 001-tdd-wprowadzenie/PHP/tests/StrongCalcTest.php
<?php
declare(strict_types=1);

namespace App\Test;

use App\StrongCalc;
use PHPUnit\Framework\TestCase;
use \Exception;

/**
 * Class StrongCalcTest
 */
class StrongCalcTest extends TestCase
{

    /**
     * @var StrongCalc
     */
    private $strongCalc;

    /**
     *
     */
    public function setUp(): void
    {
        $this->strongCalc = new StrongCalc();

        parent::setUp();
    }

    /**
     * @throws Exception
     */
    public function testCalc(): void
    {
        $this->assertEquals(6, $this->strongCalc->calc(3));
    }

    /**
     * @throws Exception
     */
    public function testNegativeStrong(): void
    {
        $this->expectException(Exception::class);
        $this->strongCalc->calc(-3);
    }
}

